components/button/floating-buttons.js

```diff
@@ -0,0 +1,207 @@
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.minHeight = '500px';
+		this._containerTop = null;
+
+		this._calcIfFloating = this._calcIfFloating.bind(this);
+		this._calcContainerPosition = this._calcContainerPosition.bind(this);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('scroll', this._calcContainerPosition);
+		window.addEventListener('resize', this._calcContainerPosition);
+		window.addEventListener('d2l-tab-panel-selected', this._calcContainerPosition);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('scroll', this._calcContainerPosition);
+		window.removeEventListener('resize', this._calcContainerPosition);
+		window.removeEventListener('d2l-tab-panel-selected', this._calcContainerPosition);
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+
+		this._containerTop = this.shadowRoot.querySelector('.d2l-floating-detection');
+
+		setTimeout(() => {
+			// need to wait a small amount for _containerTop top to be correct
+			this._calcContainerPosition();
+			this._startObserver();
+		});
+	}
+
+	updated(changedProperties) {
+		super.updated();
+		changedProperties.forEach((oldValue, propName) => {
+			if (propName === '_dir') {
+				this._calcContainerPosition();
+			}
+		});
+	}
+
+	_startObserver() {
+		const htmlElem = window.document.getElementsByTagName('html')[0];
+		this._observer = new MutationObserver((mutations) => {
+			for (let i = 0; i < mutations.length; i++) {
+				this._calcContainerPosition();
+			}
+		});
+		this._observer.observe(htmlElem, { childList: true, subtree: true });
+	}
+
+	_calcContainerPosition() {
+		this._calcIfFloating();
+		if (!this._floating) {
+			return;
+		}
+
+		const offsetParentLeft = this.offsetParent.getBoundingClientRect().left;
+		const left = this.getBoundingClientRect().left;
+		const containerLeft = left - offsetParentLeft - 1;
+		if (containerLeft !== 0) {
+			// only update this if needed - needed for firefox
+			this._containerMarginLeft = `-${containerLeft}px`;
+		}
+
+		const offsetParentRight = this.offsetParent.getBoundingClientRect().right;
+		const right = this.getBoundingClientRect().right;
+		const containerRight = offsetParentRight - right - 1;
+		if (containerRight !== 0) {
+			this._containerMarginRight = `-${containerRight}px`;
+		}
+
+		if (this.dir !== 'rtl') {
+			if (containerLeft !== 0) {
+				this._innerContainerLeft = `${containerLeft}px`;
+			}
+		} else {
+			this._innerContainerRight = `${containerLeft}px`;
+		}
+	}
+
+	_calcIfFloating() {
```

Comment:
Not sure about the method name here.  At first I thought perhaps this should be named `shouldFloat()`, but then at the end is updated the floating state if necessary, not it's more than calculating/checking if it is currently floating, or whether it should be floating.  Thoughts on renaming or refactoring?

Reply:
Yeah given that it's actually only called within `_calcContainerPosition` (there were earlier iterations where it was called in other place so it has lead to what it is now), it could just return a value and be `shouldFloat`, and we could have something like `this._floating = this._shouldFloat();`.